dxr/testing.py

```diff
@@ -305,6 +319,16 @@ def _make_file(path, filename, contents):
         file.write(contents.encode('utf-8'))
 
 
+def _format_dict_values(dictionary, *args, **kwargs):
```

Comment:
It might be simpler to just have a SingleFileTestCase.get_config() method that instantiates a dict inline and returns it. Subclasses could customize the dict by calling the superclass method and messing with the result. Plus, get_config() could take required args (like `config_dir_path`) explicitly (and document them) rather than just hoping the caller passes them in. We wouldn't need to use voodoo like deepcopy() or recursively descend through all the dict keys.

```python
    def get_config(config_dir_path):
        return {
            'DXR': {
                ...
                'temp_folder': '{config_dir_path}/temp'.format(config_dir_path),
            },
            ...
        }
```